### ISSUE TITLE
REALLY Support for max_num_results in FileSearchTool 

### ISSUE THREAD

**Issue by metaskills** [openai/openai-node#1004]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

Little to none and conflicting documentation on using max number of results.  It was added, removed, changed??? 

- https://github.com/openai/openai-node/issues/886

![Screenshot 2024-08-18 at 10 20 47 PM](https://github.com/user-attachments/assets/d3b07f4c-2ca9-4e01-8bee-3f925ca0b93a)

### To Reproduce

Try using max file results anywhere. 

- Run options `max_num_results`
- Run options `file_search.max_num_results` 
- Tools `file_search.max_num_results` 

### Code snippets

```JavaScript
None.
```

### OS

macOS

### Node version

v18

### Library version

4.56.0

**Comment by athyuttamre** [comment]
Hi @metaskills, what is the issue you're running into? 

`max_num_results` can be specified when creating a Run, documented here: https://platform.openai.com/docs/api-reference/runs/createRun#runs-createrun-tools.

The [previous issue](https://github.com/openai/openai-node/issues/886) you linked to was a bug in our OpenAPI spec, where `max_num_results` was accidentally added to the `thread.message.attachments` field.

**Comment by metaskills** [comment]
Yup, using the run options was listed in the failure list above. I can do another test to show you the output for that specific case.

**Comment by metaskills** [comment]
Also, you say documented here but noting on that page shows `max_num_results` much less in attachments?

**Comment by HAYDEN-OAI** [comment]
thanks for calling out the conflicting guidance! this will be in the next release via #1981. we added a copyable Assistants example showing `max_num_results` at `tools[].file_search.max_num_results` when creating or streaming a run, and clarified that message attachments only use `{ type: "file_search" }` to associate a file, without nested tool configuration.
